tests/integration/cluster_fixture.py

```diff
@@ -48,28 +49,87 @@ def _topic_conf(cls, conf=None):
 
         return topic_conf
 
-    def producer(self, conf=None):
+    def producer(self, conf=None, key_serializer=None, value_serializer=None):
         """
         Returns a producer bound to this cluster.
 
         :param dict conf: producer configuration overrides
+        :param Serializer value_serializer: serializer to apply to message key
+        :param Serializer key_serializer:  serializer to apply to message value
+
         :returns: a new producer instance.
         :rtype: confluent_kafka.Producer
         """
+
+        if key_serializer or value_serializer:
+            conf = self.client_conf({'key.serializer': key_serializer,
+                                     'value.serializer': value_serializer})
+            return SerializingProducer(ClientConfig(conf))
         return Producer(self.client_conf(conf))
 
-    def consumer(self, conf=None):
+    def java_producer(self, conf=None, key_serializer=None, value_serializer=None):
+        """
+        Returns a java producer handle.
+
+        :param conf:
+        :param key_serializer:
+        :param value_serializer:
+
+        :returns:
+        :return:
+        """
+        client_conf = self.client_conf(conf)
+        if key_serializer:
+            client_conf['key.serializer'] = \
+                "org.apache.kafka.common.serialization.{}".format(key_serializer)
+        if value_serializer:
+            client_conf['value.serializer'] = \
+                "org.apache.kafka.common.serialization.{}".format(value_serializer)
+
+        return JavaProducer(client_conf, self._cluster)
+
+    def consumer(self, conf=None, key_deserializer=None, value_deserializer=None):
         """
         Returns a consumer bound to this cluster.
 
+        :param Serializer key_deserializer: serializer to apply on message key
+        :param Serializer value_deserializer: serializer to apply on message value
         :param dict conf: consumer configuration overrides
+
         :returns: a new consumer instance.
-        :rtype: confluent_kafka.Consumer
+        :rtype: Consumer
         """
-        if conf is None or 'group.id' not in conf:
-            conf.update({'group.id': uuid1()})
+        consumer_conf = self.client_conf({
+            'group.id': str(uuid1()),
+            'auto.offset.reset': 'earliest'
+        })
+
+        if conf:
```

Comment:
is not None, make a habit of always checking explicitly for None since the auto-cast-bool relies on len() and other shit that is type specific.